rsfbclient-native/src/varchar.rs

```diff
@@ -0,0 +1,57 @@
+use std::{alloc, mem, ptr};
+
+#[repr(C)]
+/// Structure expected by the `fbclient`
+pub struct InnerVarchar {
+    len: u16,
+    data: [u8; 0],
+}
+
+#[derive(Debug)]
+/// Wrapper for a varchar buffer
+pub struct Varchar {
+    capacity: u16,
+    ptr: ptr::NonNull<InnerVarchar>,
+}
+
+unsafe impl Send for Varchar {}
+
+impl Varchar {
+    /// Allocate a new varchar buffer
+    pub fn new(capacity: u16) -> Self {
+        let ptr = ptr::NonNull::new(unsafe {
+            alloc::alloc_zeroed(layout(capacity as usize)) as *mut InnerVarchar
```

Comment:
It's required that this allocation should be zeroed ? 

If don't, zeroing would be useless as that memory allocated here will be overwrite soon after this, Right?

Also a further improvement would be using a memory pool or memory zone for allocation for reducing the alloc/free overhead.

Reply:
I don't really know if the allocation needs to be zeroed, so i did it to eliminate the possibility of reading uninitialized memory from safe code, else the `as_bytes` function would need to be `unsafe`.

About the free / alloc, the `Varchar` is allocated only once per statement and reused to all rows, so i don't think it needs more optimization, but can be done in another pr maybe.

Reply:
I've initialized only the `len` field to 0, so that it will be safe to call `as_bytes`.